python/pyspark/sql/streaming/listener.py

```diff
@@ -477,7 +477,7 @@ def fromJson(cls, j: Dict[str, Any]) -> "StreamingQueryProgress":
             name=j["name"],
             timestamp=j["timestamp"],
             batchId=j["batchId"],
-            batchDuration=j["batchDuration"],
+            batchDuration=j["batchDuration"] if "batchDuration" in j else None,
```

Comment:
I'm wondering when Spark receive a JSON file without `batchDuration`? We are inside the same Spark versions, aren't we? Is this listener able to listen old Spark?

Reply:
So this is currently how this method used: in spark connect, the way the listener works, is that 
1. The user's listener code is serialized and sent to spark server
2. The server starts a scala listener, [in which starts a python process](https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/StreamingQueryListenerHelper.scala) (essentially as another connect client), that runs the user's code
3. Each time a new event comes in, the event on [java side is serialized to json](https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/StreamingQueryListenerHelper.scala#L52) and passed to server python process, which calls this `fromJson` [method to convert it back to the actual `StreamingQueryProgress` object](https://github.com/apache/spark/blob/master/python/pyspark/sql/connect/streaming/worker/listener_worker.py#L76-L77)

But before https://github.com/apache/spark/pull/42077 in 3.5, that field is not added in the jvm `json` method of `StreamingQueryProgress`. Here it excepts that to always be presented, hence we get an error